### PR TITLE
PLANNER-552: Workbench domain editor: visually show which fields are planning variables

### PR DESCRIPTION
**Pull request by mcimbora**
The PR adds support for adding domain-specific icons to a data object table. This is particularly useful in situations, when domain-specific functionality (e.g. planner field annotations) is hidden in docks, as it avoids the need to click on every row of the table to find the affected fields.

For this purpose I reused the column which formerly contained "i" symbol, i.e. a link to another type displayed in case the field is a complex type. I replaced this functionality by clickable "type" column (still applies to complex types only).

Individual domains can define field metadata by implementing FieldMetadataProvider interface. Data modeller uses CDI to discover all domain implementations.

Part of:
- https://github.com/kiegroup/kie-wb-common/pull/1078
- https://github.com/kiegroup/optaplanner-wb/pull/203

@wmedvede As this affects the data modeller, can you please take a look at the PRs? Thank you in advance.

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/model/editor/FieldMetadata.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.workbench.common.screens.datamodeller.model.editor;
+
+import org.jboss.errai.common.client.api.annotations.Portable;
+
+@Portable
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/model/editor/FieldMetadata.java]: Is it portable? Isn't this just client-side?

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/model/editor/ImageWrapper.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.workbench.common.screens.datamodeller.model.editor;
+
+import org.jboss.errai.common.client.api.annotations.Portable;
+
+@Portable
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-api/src/main/java/org/kie/workbench/common/screens/datamodeller/model/editor/ImageWrapper.java]: Is this _really_ portable too?

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/editor/DataObjectBrowserViewImpl.java [hunk 3]
```diff
@@ -89,9 +96,11 @@
     @UiField(provided = true)
     SimpleTable<ObjectProperty> propertiesTable = new BrowserSimpleTable<ObjectProperty>(1000);
 
-    @Inject
+    //    @Inject
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/editor/DataObjectBrowserViewImpl.java]: Not sure you want that!

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/editor/DataObjectBrowserViewImpl.java [hunk 4]
```diff
@@ -272,50 +283,63 @@ public String getValue(ObjectProperty objectProperty) {
...
-                    return null;
+            public List<ImageWrapper> getValue(final ObjectProperty property) {
+                List<ImageWrapper> imageWrapperList = fieldMetadataProviderList.stream()
+                        .map(p -> p.getFieldMetadata(property))
+                        .flatMap(s -> s.map(Stream::of).orElseGet(Stream::empty))
+                        .map(FieldMetadata::getImageWrapper)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList());
+                if (!imageWrapperList.isEmpty()) {
+                    int targetColumnWidth = imageWrapperList.size() * 30;
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/editor/DataObjectBrowserViewImpl.java]: Doesn't this assume icons are 30px wide?

Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/editor/DataObjectBrowserViewImpl.java]: Or use ```MultiImageCell.IMAGE_WIDTH``` (after making it public)?

**Comment by mcimbora** [comment]
@manstis Haha, just wondering how you manage to do so many things at the same moment. I suspect you're either a bot or a small team of programmers. Thanks for the review.

**Comment by ge0ffrey** [comment]
```
4:06:48 [INFO]       [ERROR] Errors in 'jar:file:/home/jenkins/.m2/repository/com/google/jsinterop/base/1.0.0-beta-1/base-1.0.0-beta-1.jar!/jsinterop/base/InternalJsUtil.java'
14:06:48 [INFO]          [ERROR] Line 225: Referencing field 'java.lang.Class.jsConstructor': unable to resolve field in class 'java.lang.Class'
14:06:50 [INFO]    [ERROR] Aborting compile due to errors in some input files
```

**Comment by mcimbora** [comment]
Jenkins retest this.

**Pull request merged**